packages/pwa-storefront-mcp/src/utils/create-new-component-tool.js

```diff
@@ -0,0 +1,342 @@
+        this.reset()
+
+        return messages.join('\n')
+    }
+
+    /**
+     * Resets the tool state for the next component creation
+     */
+    reset() {
+        this.currentStep = 0
+        this.componentData = {
+            name: null,
+            location: null,
+            entityType: null
+        }
+    }
+
+    /**
+     * Creates a new React component file.
+     * @param {string} componentName - Name for the new component.
+     * @param {string} projectDir - The absolute path to the project directory for the new component.
+     */
+    async createComponentFile(componentName, projectDir) {
+        const kebabDirName = toKebabCase(componentName)
+        const pascalComponentName = toPascalCase(componentName)
+        const componentDir = path.join(projectDir, kebabDirName)
+        try {
+            await fs.mkdir(componentDir, {recursive: true})
+            // Create component file
+            const componentFilePath = path.join(componentDir, 'index.jsx')
+            const codeToWrite = `${getCopyrightHeader()}
+import React from 'react';
+
+const ${pascalComponentName} = () => {
+  return (
+    <div>${pascalComponentName} component</div>
+  );
+};
+
+export default ${pascalComponentName};
+`
+            await fs.writeFile(componentFilePath, codeToWrite, 'utf-8')
+            return `✅ Created ${componentFilePath}`
+        } catch (err) {
+            console.error('Error during file creation:', err)
+            return `❌ Error creating component file at ${componentDir}: ${err.message}`
+        }
+    }
+
+    /**
+     * Updates the component file to be a presentational component for the given data model.
+     * @param {string} entityType - The entity type (e.g., 'product').
+     * @param {string} componentName - The component name.
+     * @param {string} location - The absolute path to the component's parent directory.
+     * @param {object} dataModel - The data model schema (properties object).
+     */
+    async updateComponentToPresentational(
+        entityType,
+        componentName,
+        location,
+        dataModel,
+        options = {}
+    ) {
+        const kebabDirName = toKebabCase(componentName)
+        const pascalComponentName = toPascalCase(componentName)
+        const componentDir = path.join(location, kebabDirName)
+        await fs.mkdir(componentDir, {recursive: true})
+        const componentFilePath = path.join(componentDir, 'index.jsx')
+        const fields = Object.keys(dataModel)
+        let code = ''
+
+        // Special logic for product entity
+        if (entityType === 'product') {
+            // If options.list is true, generate a list-of-products component
+            if (options.list) {
+                code = `${getCopyrightHeader()}
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Box, Text, Image, Stack } from '@chakra-ui/react';
+
+const ${pascalComponentName} = ({ products }) => (
+    <Stack spacing={4}>
```

Comment:
Note for PWA Kit team - we'll want to update these components to Chakra v3 as part of migrating this to v4.